### PR TITLE
Add `hs create website-theme <dest>` command

### LINKED ISSUE #54: Remove git requirement for @hubspot/create-cms-project
We should remove the need to have `git` installed to run `@hubspot/create-cms-project`, the created project should not be connected to the https://github.com/HubSpot/cms-theme-boilerplate repo

### PR DESCRIPTION
**Pull request by corymartin**
Fixes #54

- Adds command `hs create website-theme <dest>`
- Does not use git but rather downloads zip archive.
  - Currently using master branch.
  - Need to use latest release when there is one.  Presently there is only a draft and no releases.
     - https://github.com/HubSpot/hubspot-cms-tools/issues/54#issuecomment-577836161
     - https://github.com/HubSpot/cms-theme-boilerplate/releases/tag/untagged-b795927be7624c07ddf4
- Adds on option to fetch a specified release version (the tag specified in the [release](https://github.com/HubSpot/cms-theme-boilerplate/releases)). If omitted then the latest release is fetched.
  - `--theme-version=v0.1` . Will also work without `"v"` e.g. `--theme-version=0.1`

#### Example
```
$ hs create website-theme ./my-theme

Fetching Boilerplate v0.1...
Completed theme fetch.
Extracting theme source...
Completed theme source extraction.
Copying theme source...
Completed copying theme source.
Success: your new website-theme project has been created in /Users/devperson/my-theme
```

Results in
```
- /my-theme
  + /css
  + /images
  - /js
    - main.js
  + /modules
  + /templates
```

/cc @williamspiro

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -2901,6 +2901,16 @@ extglob@^2.0.4:
     snapdragon "^0.8.1"
     to-regex "^3.0.1"
 
+extract-zip@^1.6.7:
+  version "1.6.7"
+  resolved "https://npm.hubteam.com/npm-nexus/repository/npm-all/extract-zip/-/extract-zip-1.6.7.tgz#a840b4b8af6403264c8db57f4f1a74333ef81fe9"
```
Comment (corymartin) [inline yarn.lock]: oops

**Comment by williamspiro** [comment]
Thanks! Could we align the final success message with https://github.com/HubSpot/hubspot-cms-tools/pull/56

cc @iamw11y

File: packages/cms-lib/themes.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+const ZIP_CONTENT_TYPE = 'application/zip';
+const TMP_BOILERPLATE_FOLDER_PREFIX = 'hubspot-cms-theme-boilerplate-';
+
+/**
+ * @returns {Buffer} Zip data buffer
+ */
+async function downloadCmsThemeBoilerplate() {
+  try {
+    logger.log('Fetching theme...');
+    const zip = await request.get(THEME_BOILERPLATE_ZIP_URI, {
```
Comment (miketalley) [inline packages/cms-lib/themes.js]: To be consistent here, can we import and use `cms-lib/http.js`'s `.get`?

Comment (corymartin) [reply packages/cms-lib/themes.js]: That encapsulates HubSpot APIs with auth to make that process easier.
Here we're just downloading a zip from a public URI.

File: packages/cms-lib/themes.js [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+ * @param {String} tag - Git tag to fetch for. If omitted latest will be fetched.
+ * @returns {Buffer|Null} Zip data buffer
+ */
+async function downloadCmsThemeBoilerplate(/*tag = null*/) {
+  try {
+    // const releaseData = fetchReleaseData(tag);
+    // if (!releaseData) return;
+    // const { zipball_url: zipUrl, tag_name: tagName } = releaseData;
+    // logger.log(`Fetching theme ${tagName}...`);
+    // TODO: Use `zipUrl` when we start releases.
```
Comment (corymartin) [inline packages/cms-lib/themes.js]: @gcorne @axellabs When could we cut a release of the boilerplate so this code can be used? There is only a draft release presently.
https://github.com/HubSpot/cms-theme-boilerplate/releases

**Comment by gcorne** [comment]
I haven't had a chance to dive too deeply into the zip extraction code, but I do think we should test it out on Windows.

**Comment by corymartin** [comment]
> I haven't had a chance to dive too deeply into the zip extraction code, but I do think we should test it out on Windows if that hasn't been done already.

👍 

I also want to try a different extraction package that allows direct extract of buffer data instead of first having to write the zip data to disk.
https://www.npmjs.com/package/yauzl

File: packages/cms-lib/themes.js (hunk 3 above)
Comment (axellabs) [reply packages/cms-lib/themes.js]: I'm on it right now!

Comment (axellabs) [reply packages/cms-lib/themes.js]: The release is out

Comment (corymartin) [reply packages/cms-lib/themes.js]: Thanks @axellabs 
https://api.github.com/repos/HubSpot/cms-theme-boilerplate/releases/latest

**Comment by corymartin** [comment]
Screen Shot 2020-01-28 at 14 05 44

/cc @gcorne

**Comment by williamspiro** [comment]
This is awesome @corymartin !

**Comment by corymartin** [comment]
@gcorne The boilerplate doesn't include `fields.json` and `theme.json`.  Should we stub those files out and populate the "label" field?

**Comment by gcorne** [comment]
> @gcorne The boilerplate doesn't include fields.json and theme.json. Should we stub those files out and populate the "label" field?

The boilerplate will become a proper theme when "themes" are generally available. See https://github.com/HubSpot/cms-theme-boilerplate/pull/71

**Pull request merged**